### PR TITLE
Pass the Webpack options.filename to the preprocessor function.

### PR DESCRIPTION
**Pull request by soluml**
## Motivation

Solves issue #571.

## Test plan

Updated types to reflect the addition of a third argument. No real functionality changes, just passing in additional data which was provided by the bundler.

### CODE REVIEW

File: src/transform.ts [hunk 1]
```diff
@@ -133,7 +133,11 @@ export default function transform(code: string, options: Options): Result {
     });
 
     // Run each rule through stylis to support nesting
-    cssText += `${preprocessor(selector, rules[selector].cssText)}\n`;
+    cssText += `${preprocessor(
+      selector,
+      rules[selector].cssText,
+      options.filename
```
Comment (satya164) [inline src/transform.ts:139]: Can you make this an object instead? It will be more futureproof if there's need for more metadata

Comment (soluml) [reply src/transform.ts:139]: Sure! That's a good idea. Instead of just filename I'll pass the entire options object as a 3rd argument.

Comment (soluml) [reply src/transform.ts]: Done

**Comment by jayu** [comment]
I treat this PR as stale, I will reopen if @soluml decide to update!

**Pull request closed without merging**